### PR TITLE
`cpp_register(quiet=)` set to TRUE on non-interactive session (incl testthat runs)

### PR DESCRIPTION
**Pull request by romainfrancois**
So that e.g. the testthat output is less verbose: 

image

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 * `cpp_source()` errors on non-existent file (#261). 
 
+* `cpp_register()` is quiet by default when R is non interactive.
```
Comment (DavisVaughan) [inline NEWS.md:9]:
```suggestion
* `cpp_register()` is quiet by default when R is non interactive (#289).
```

File: R/source.R [hunk 2]
```diff
@@ -110,7 +110,7 @@ cpp_source <- function(file, code = NULL, env = parent.frame(), clean = TRUE, qu
   check_valid_attributes(all_decorations, file = orig_file_path)
 
   cli_suppress(
-    funs <- get_registered_functions(all_decorations, "cpp11::register")
+    funs <- get_registered_functions(all_decorations, "cpp11::register", quiet = quiet)
   )
```
Comment (DavisVaughan) [inline R/source.R:114]: Was `cli_suppress()` already essentially forcing `quiet = TRUE`?

i.e. if you call `cpp11_source(quiet = FALSE)` right now, can you even force output to be shown? If not, I think we should think about removing `cli_suppress()` altogether and just hardcoding this to `quiet = TRUE`

File: tests/testthat/test-register.R [hunk 3]
```diff
@@ -614,12 +614,18 @@ extern \"C\" attribute_visible void R_init_testPkg(DllInfo* dll){
...
     dir.create(file.path(p, "src"))
     file.copy(test_path("single.cpp"), file.path(p, "src", "single.cpp"))
-    expect_message(cpp_register(p), "1 functions decorated with [[cpp11::register]]", fixed = TRUE)
+    suppressMessages(
+      # cpp_register() sends 3 messages, but expect_message() only checks
+      # for one, so suppress the other ones, so that they don't appear in the
+      # testthat output
+      expect_message(
+        cpp_register(p, quiet = FALSE), "1 functions decorated with [[cpp11::register]]", fixed = TRUE)
+    )
```
Comment (DavisVaughan) [inline tests/testthat/test-register.R:628]: What if you just do `expect_snapshot()` to capture everything from `cpp11_register(p, quiet = FALSE)`?

**Review by DavisVaughan — APPROVED** [review]

File: R/source.R (hunk 2 above)
Comment (romainfrancois) [reply R/source.R]: I guess now we can even remove `cli_suppress()` from the package.

File: tests/testthat/test-register.R (hunk 3 above)
Comment (romainfrancois) [reply tests/testthat/test-register.R:625]: Thanks, that's much better. I've opened #291 to follow up for other use cases.

**Pull request merged**